src/netconf_subscribed_notifications.c

```diff
@@ -601,22 +839,25 @@ csn_config_sub_receivers_prepare(const struct lyd_node *node_receiver,
 
     receiver.name = strdup(lyd_get_value(node_receiver_name));
     if (!receiver.name) {
-        rc = SR_ERR_NO_MEMORY;
+        EMEM;
         csn_receiver_destroy(&receiver, 0);
```

Comment:
Seems you can put this into the cleanup with condition `if (rc) ...` and save this call in all following error conditions.